hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,256 @@
+---
+
+# Overview
+
+This document specifies the functionality of the OpenTitan power manager.
+
+## Features
+
+- Cold boot, low power entry / exit and reset support.
+- 2 different low power modes.
+- Software initiated low power entry and hardware requested low power exit.
+- Peripheral reset requests
+- Low power abort and low power fall-through support.
+
+## Description
+
+The power manager sequences power, clocks, and reset resources of the design through cold boot, low power entry/exit and reset scenarios.
+
+Cold boot, also known as POR (power on reset) is the first reset state of the design.
+The power manager sequences the design from a freshly reset state to an active state where software can be initialized.
+
+- Low power entry is the process in which the device enters one of two low power modes (sleep or deep sleep).
+- Low power exit is the process in which the device exits low power mode and returns to active state.
+- Low power entry is always initiated by software, while low power exit is always initiated by a previously setup hardware event such as pins or internal timers.
+- The power manager processes the software and hardware requests to perform the appropriate actions.
+
+Reset scenarios refer to non-POR events that cause the device to reboot.
+There are various stimuli that can cause such a reset, ranging from external user input to watchdog timeout.
+The power manager processes the reset request and brings the device to an appropriate state.
+
+
+# Theory Of Operation
+
+The power manager performs the following functions:
+- Turn on/off power domain(s).
+- Control root resets with the reset manager.
+- Control root clock enables with AST and clock manager.
+- Sequence various power up activities such as OTP sensing, life cycle initiation and releasing software to execute.
+
+If there is a need to perform ROM hashing as an integrity check in the future, that will also be managed by the power manager.
+
+## Block Diagram
+
+See the below high level block diagram that illustrates the connections between the power manager and various system components.
+Blocks outlined with a solid magenta line are always on; while blocks outlined with a dashed magenta line are a mix of components that are and those that are not.
+
+![Power Manager Connectivity Diagram](pwrmgr_connectivity.svg)
+
+
+## Overall Sequencing
+
+The power manager contains two state machines.
+One operates on the always-on slow clock (this clock is always running and usually measured in KHz) and is responsible for turning faster clocks on and off and managing the power domains.
+The other operates on a normal fixed clock (usually measured in MHz) and is responsible for everything else in the power sequence.
+
+The following diagram breaks down the general functionality of both.
+The state machines are colored based on their clock domains.
+The green state machine is clocked by the normal fixed domain, while the organge state machine is clocked by the slow domain.
```

Comment:
Typo: "organge"

Reply:
i was obviously going for orangutan there...